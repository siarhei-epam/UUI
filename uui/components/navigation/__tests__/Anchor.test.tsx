import React from 'react';
import renderer from 'react-test-renderer';
import { Anchor } from "../Anchor";

describe('Anchor', () => {
    it('should rendered correctly', () => {
        const tree = renderer.create(<Anchor>Test Anchor</Anchor>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should rendered with props correctly', () => {
        const tree = renderer.create(<Anchor href="https://uui.epam.com" isDisabled={ true }>Test Anchor</Anchor>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});