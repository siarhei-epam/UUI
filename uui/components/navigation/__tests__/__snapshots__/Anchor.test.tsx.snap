// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor should rendered correctly 1`] = `
<a
  className="container uui-anchor uui-enabled -clickable"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="link"
  tabIndex={0}
>
  Test Anchor
</a>
`;

exports[`Anchor should rendered with props correctly 1`] = `
<a
  aria-disabled={true}
  className="container uui-anchor uui-disabled -clickable"
  disabled={true}
  href="https://uui.epam.com"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="link"
  tabIndex={-1}
>
  Test Anchor
</a>
`;
